### PR TITLE
Ignore `MIX_TEST_PARTITION` when partitions set to 1

### PR DESCRIPTION
**Pull request by hauleth**
This is meant to make GitLab CI integration with partitions much easier as will allow to use

```shell
MIX_TEST_PARTITION=${CI_NODE_INDEX} mix test --cover --partitions ${CI_NODE_TOTAL}
```

As a test task instead of additionally taking care of case when the parallel testing is disabled.

### CODE REVIEW

**Comment by hauleth** [comment]
Just FYI, it wasn't much harder to make it work in current state, as all was needed was to use Bash "fallback if not set" syntax `:-1`:

```bash
MIX_TEST_PARTITION=${CI_NODE_INDEX:-1} mix test --cover --partitions ${CI_NODE_TOTAL}
```

But I think that proposed API makes it easier to work with.

File: lib/mix/lib/mix/tasks/test.ex [hunk 1]
```diff
@@ -486,7 +486,7 @@ defmodule Mix.Tasks.Test do
     cover =
       if opts[:cover] do
         compile_path = Mix.Project.compile_path(project)
-        partition = opts[:partitions] && System.get_env("MIX_TEST_PARTITION")
+        partition = opts[:partitions] > 1 && System.get_env("MIX_TEST_PARTITION")
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/test.ex:489]: Maybe we should Delete the option if it is set to 1 so we don’t have to handle it elsewhere?

**Comment by josevalim** [comment]
Looks great! Just one comment and please add a mention to the docs and we are good to go!

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**